### PR TITLE
JBPM-7088: Stunner - Palette re-design iteration #2

### PR DESCRIPTION
**Pull request by wmedvede**

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello @romartin @hasys , would you mind review this PR?
Thanks!

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/resources/i18n/DefaultWorkbenchConstants.java [hunk 1]
```diff
@@ -151,6 +151,8 @@
 
     String StunnerDesigner();
 
+    String StunnerDesignerPreferences();
```
Comment (jomarko) [inline kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/resources/i18n/DefaultWorkbenchConstants.java]: Could we rename also the constant above as it is also used in admin perspective?  I think we should not refer the same thing with two different names. Let me know what do you think.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/BS3PaletteWidgetImpl.java [hunk 2]
```diff
@@ -264,6 +285,17 @@ private void onPaletteItemMouseDown(final String id,
                            getIconSize());
     }
 
+    private void onOpenCategory(String categoryId) {
+        categoryWidgets.entrySet().stream().filter(entry -> !entry.getKey().equals(categoryId)).forEach(entry -> entry.getValue().setVisible(false));
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/BS3PaletteWidgetImpl.java]: Streams used in proper way, however I would propose slight improvement:

```java
categoryWidgets.entrySet()
                .stream()
                .filter(entry -> !Objects.equals(entry.getKey(), categoryId))
                .forEach(entry -> entry.getValue().setVisible(false));
```

I think proposed formatting is more readable and prevents NPE in filter part, let me know WDYT.

File: kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/resources/i18n/DefaultWorkbenchConstants.java (hunk 1 above)
Comment (wmedvede) [reply kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/resources/i18n/DefaultWorkbenchConstants.java]: hi  @jomarko , sorry but I don't understand this suggestion.

**Comment by jomarko** [comment]
@wmedvede sorry I was not clear enough, will try to clarify.

Your PR adds Stunner item to admin perspective, right? Stunner is the new editor for business processes. However this new editor for business processes can be enabled/disabled in other admin perspective, see the attached picture, however in that perspective I refer to, it is named **(New) jBPM Designer Editor**, so my question is, if we shouldn't consolidate these two different names of the same thing. Am I clear more now please? Please let me know or reach on IRC.

screenshot from 2018-04-18 09-50-44

**Comment by manstis** [comment]
@wmedvede @jomarko Please be aware @romartin has a JIRA to enable the new Designer by default and disable the old Designer; so the inconsistency in names may not be an issue after that is complete.

**Comment by jomarko** [comment]
well so ignore my comment please

**Comment by wmedvede** [comment]
Hi @jomarko , thanks now I understand. I didn't know about this other place for enabling/disabling Stunner. But yes, I agree. It doesn't look good to see "(New) jBPM Designer Editor" in one place and "Stunner" in the other place.
Let's see what @romartin says about his JIRA, but yes, if the ability of enabling/disabling Stunner is removed, then probably the naming inconsistency goes by with the JIRA. If not we can make to use the same constant in both places, and in this last case I'd vote for just show "Stunner" in both places rather than "(New) jBPM Designer Editor".

**Comment by wmedvede** [comment]
@manstis yes you are right, well, I'll be happy to add whatever more appropriated text ;). Probably just "Process Designer"? @romartin wdyt?

**Comment by domhanak** [comment]
Jenkins execute full downstream build

**Comment by domhanak** [comment]
@wmedvede Hello, I tried the BPMN part in Firefox. IE11 and Chrome. Looks very good.
I found only this:
-  In IE11 this is happening with the palette, it was not happening before
image
- In IE11 While dragging an item from palette, no icon is shown during dragging as it is shown in Firefox, Chrome

**Comment by hasys** [comment]
Hi @wmedvede,

You can install IE11 on Windows 10:
https://docs.microsoft.com/en-us/microsoft-edge/deploy/enterprise-guidance-using-microsoft-edge-and-ie11

**Comment by wmedvede** [comment]
Hi @romartin , @jomarko 's suggestions were added, and also instead of "Stunner" we will now show "Process Designer" as we spoke. 
PR was also rebased with master

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by domhanak** [comment]
-  I did find out one more thing, with sub-processes not showing icons for Event and Embedded ones.
@wmedvede  This is however happening in Firefox.
image

**Comment by wmedvede** [comment]
@domhanak hmm, it works well in my local. Could you please verify/try to clear your browser cache just in case it might be related with cached data?

**Comment by wmedvede** [comment]
Jenkins execute full downstream build

**Comment by domhanak** [comment]
@wmedvede Sure I will try it again as you requested

**Comment by domhanak** [comment]
@wmedvede I tried it a couple times and it seems that I can no longer reproduce it  :)

**Comment by wmedvede** [comment]
Jenkins execute full downstream build

**Comment by wmedvede** [comment]
@romartin PR was updated to solve the IE11 alignment issue reported above. As we spoke this afternoon the other issues related with the toolbox and the dragging of the icons are being treated separately.
The PR was also rebased with master due to merge conflicts produced by other PRs merged in the middle.

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by domhanak** [comment]
@wmedvede Hello, I can no longer reproduce the issues in iE11. Thanks!
One small thing - I can't see the icon of node I selected next to my cursor, when adding it onto canvas, is it palette related or not?

**Comment by wmedvede** [comment]
@domhanak Hello, this "dragging icon" issue belongs let's say to the "IE11 issues Chapter" but is not related with the palette nor this PR. We are aware of this issue and some other related with the toolbox, but they will be treated separately. 
Thanks.

**Comment by romartin** [comment]
sorry @wmedvede this needs to be rebased for merging

**Comment by wmedvede** [comment]
@romartin Rebased. Please try to merge this, it takes years to rebase, solve conflicts, and rebuild and check all again, and over again. Specially with jenkkins and the maven repositories outdated.

**Pull request merged**